cos-fleetshard-sync/src/main/java/org/bf2/cos/fleetshard/sync/housekeeping/reapers/AddonReaper.java

```diff
@@ -0,0 +1,183 @@
+package org.bf2.cos.fleetshard.sync.housekeeping.reapers;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.apache.http.annotation.Obsolete;
+import org.bf2.cos.fleetshard.support.Service;
+import org.bf2.cos.fleetshard.support.resources.Resources;
+import org.bf2.cos.fleetshard.support.watch.AbstractWatcher;
+import org.bf2.cos.fleetshard.sync.FleetShardSync;
+import org.bf2.cos.fleetshard.sync.FleetShardSyncConfig;
+import org.bf2.cos.fleetshard.sync.housekeeping.Housekeeper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.Namespace;
+import io.fabric8.kubernetes.api.model.NamespaceList;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.Watch;
+import io.fabric8.kubernetes.client.dsl.FilterWatchListDeletable;
+import io.fabric8.kubernetes.client.dsl.base.ResourceDefinitionContext;
+
+@ApplicationScoped
+public class AddonReaper implements Housekeeper.Task, Service {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AddonReaper.class);
+    public static final String ID = "addon.cleanup";
+
+    private final KubernetesClient kubernetesClient;
+    private final FleetShardSyncConfig config;
+    private final FleetShardSync fleetShardSync;
+    private final ConfigMapWatcher watcher;
+    private final AtomicLong retries;
+    private final AtomicBoolean running;
+    private final AtomicBoolean taskRunning;
+
+    private Watch watcherInstance;
+
+    public AddonReaper(KubernetesClient kubernetesClient, FleetShardSyncConfig config, FleetShardSync fleetShardSync) {
+        this.kubernetesClient = kubernetesClient;
+        this.config = config;
+        this.fleetShardSync = fleetShardSync;
+        this.watcher = new ConfigMapWatcher();
+        this.retries = new AtomicLong(0);
+        this.running = new AtomicBoolean();
+        this.taskRunning = new AtomicBoolean();
+    }
+
+    @Override
+    public String getId() {
+        return ID;
+    }
+
+    @Obsolete
+    public void start() throws Exception {
```

Comment:
Shouldn't this just create an instance of ConfigMapWatcher? (or start the one instantiated in the constructor)

Reply:
good catch (part 2)